src/OrbitGl/CaptureViewElement.cpp

```diff
@@ -60,6 +60,17 @@ CaptureViewElement::EventResult CaptureViewElement::OnMouseWheel(
   return EventResult::kIgnored;
 }
 
+void CaptureViewElement::OnMouseMove(const Vec2& mouse_pos) {
```

Comment:
Actually, that was probably the reason why you were passing mouse move elements to all children, not only the one underneath the mouse, right? I didn't fully consider that, and I think it's a bit more complicated than would appear initially. I still don't think the current solution is the right one, it feels hacky - events shouldn't be handled that differently, and it's surprising if the move events are passed to all children.

Reply:
I pass MouseMoved event just for OnMouseLeave.

I also thought that it would be nice to have this updated in CaptureViewElements.
```
Vec2 mouse_pos_last_click_;
Vec2 mouse_pos_cur_;
```

But I agree that we can process MouseMove with at most 2 events, one Enter and one Leave. Maybe TimeGraph could save the element pointed by the mouse to tell it Leave() after the mouse went out?

Reply:
> It would be safer to create mouse events for enter and leave during DoUpdateLayout in case the state changes instead of this special handling in MouseMove, WDYT?

I also thought about that, but luckily we don't have buttons changing positions for now. With your solution you mean storing mouse position changed in TimeGraph and emitting the event later after updating positions?

Reply:
After a offline talk, we decided that the parent will redirect events to the correspondent child and not worrying for now to updated positions on Layout.